libkpod/image/image.go

```diff
@@ -212,32 +210,55 @@ func findImageInSlice(images []storage.Image, ref string) (storage.Image, error)
 	return storage.Image{}, errors.New("could not find image")
 }
 
-// DigestAndSize returns the size of the image in the given store and the
-// digest of its manifest, if it has one, or "" if it doesn't.
-func DigestAndSize(store storage.Store, img storage.Image) (digest.Digest, int64, error) {
+// InfoAndDigestAndSize returns the inspection info and size of the image in the given
+// store and the digest of its manifest, if it has one, or "" if it doesn't.
+func InfoAndDigestAndSize(store storage.Store, img storage.Image) (*types.ImageInspectInfo, digest.Digest, int64, error) {
+	_, _, info, sum, size, err := annotationsAndConfigAndInfoAndDigestAndSize(store, img)
+	return info, sum, size, err
+}
+
+func annotationsAndConfigAndInfoAndDigestAndSize(store storage.Store, img storage.Image) (map[string]string, *ociv1.Image, *types.ImageInspectInfo, digest.Digest, int64, error) {
```

Comment:
This is parsing a lot of information at once.  Would it make sense  to split this into multiple functions?  Or to have this one function call out to other functions so you only have to get the imgRef once?